src/server/rdb_save.cc

```diff
@@ -1522,4 +1532,37 @@ void RdbSerializer::CompressBlob() {
   ++compression_stats_->compressed_blobs;
 }
 
+RestoreSerializer::RestoreSerializer(CompressionMode compression_mode)
+    : SerializerBase(compression_mode) {
+}
+
+error_code RestoreSerializer::SaveEntry(const PrimeKey& pk, const PrimeValue& pv,
+                                        uint64_t expire_ms, DbIndex dbid) {
+  absl::InlinedVector<string_view, 4> args;
+
+  string key_buffer;
+  string_view key = pk.GetSlice(&key_buffer);
+  args.push_back(key);
+
+  string expire_str = absl::StrCat(expire_ms);
+  args.push_back(expire_str);
+
+  string value = RdbSerializer::DumpObject(pv);
```

Comment:
please see how you can change the api of DumpObject to pass it a buffer that you will stor as a member of this class so you wont do so many string allocations while we serialize all entries in database